File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        diff = -1
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[j] > nums[i]:
                    diff = max(diff, nums[j] - nums[i])
        return diff if diff != -1 else -1

File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int globalmin = nums[0];
        int maxdiff = -1;
        for (int i = 0; i < nums.size(); i++) {
            maxdiff = max(maxdiff, nums[i] - globalmin);
            globalmin = min(globalmin, nums[i]);
        }
        if (maxdiff) {
           return maxdiff; 
        }
        return -1;
    }
};